### PR TITLE
feat: add a http proxy with api gateway typescript example

### PR DESCRIPTION
**Pull request by syeutyu**
I'm adding an example here that deploys http proxy to setup using API gateway.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

**Comment by richardhboyd** [comment]
The Readme should have more information about what this example does and when it would be used. Links to the APIGW documentation for proxies would also be helpful.

**Comment by syeutyu** [comment]
if I need more information on this example or made some typo
please let me know what should I do :)

File: README.md [hunk 1]
```diff
@@ -49,6 +49,7 @@ $ cdk destroy
 | [ecs-service-with-logging](https://github.com/aws-samples/aws-cdk-examples/tree/master/typescript/ecs/ecs-service-with-logging/) | Starting a container fronted by a load balancer on ECS |
 | [fargate-service-with-logging](https://github.com/aws-samples/aws-cdk-examples/tree/master/typescript/ecs/fargate-service-with-logging/) | Starting a container fronted by a load balancer on Fargate |
 | [custom-logical-names](https://github.com/aws-samples/aws-cdk-examples/tree/master/typescript/custom-logical-names/) | Example of how to override logical name allocation |
+| [proxy-apigateway](https://github.com/aws-samples/aws-cdk-examples/tree/master/typescript/proxy-apigateway/) | Use ApiGateway to set up http proxy |
```
Comment (nija-at) [inline README.md:52]: Rename this 'http proxy apigateway' everywhere to disambiguate it from a lambda proxy.

Comment (syeutyu) [reply README.md:52]: I don't understand what you mean if my thinking is corrected
Do you mean more specific about the difference between the'http proxy apigateway' and the'lambda proxy'?

If you don't mind could you see me some example?

Comment (nija-at) [reply README.md:52]: Change the name to 'http-proxy-apigateway'

**Comment by syeutyu** [comment]
I fixed it!

File: README.md [hunk 2]
```diff
@@ -49,6 +49,7 @@ $ cdk destroy
 | [ecs-service-with-logging](https://github.com/aws-samples/aws-cdk-examples/tree/master/typescript/ecs/ecs-service-with-logging/) | Starting a container fronted by a load balancer on ECS |
 | [fargate-service-with-logging](https://github.com/aws-samples/aws-cdk-examples/tree/master/typescript/ecs/fargate-service-with-logging/) | Starting a container fronted by a load balancer on Fargate |
 | [custom-logical-names](https://github.com/aws-samples/aws-cdk-examples/tree/master/typescript/custom-logical-names/) | Example of how to override logical name allocation |
+| [http-proxy-apigateway](https://github.com/aws-samples/aws-cdk-examples/tree/master/typescript/proxy-apigateway/) | Use ApiGateway to set up lambda proxy |
```
Comment (nija-at) [inline README.md:52]: As far as I can tell, this isn't setting up a lambda proxy, but a http proxy

```suggestion
| [http-proxy-apigateway](https://github.com/aws-samples/aws-cdk-examples/tree/master/typescript/proxy-apigateway/) | Use ApiGateway to set up a http proxy |
```

Comment (syeutyu) [reply README.md]: I agree with you, that is my typo

**Comment by syeutyu** [comment]
I have changed a lambda proxy everywhere to http proxy.
Could you check this pr one more?

File: README.md [hunk 3]
```diff
@@ -49,6 +49,7 @@ $ cdk destroy
 | [ecs-service-with-logging](https://github.com/aws-samples/aws-cdk-examples/tree/master/typescript/ecs/ecs-service-with-logging/) | Starting a container fronted by a load balancer on ECS |
 | [fargate-service-with-logging](https://github.com/aws-samples/aws-cdk-examples/tree/master/typescript/ecs/fargate-service-with-logging/) | Starting a container fronted by a load balancer on Fargate |
 | [custom-logical-names](https://github.com/aws-samples/aws-cdk-examples/tree/master/typescript/custom-logical-names/) | Example of how to override logical name allocation |
+| [http-proxy-apigateway](https://github.com/aws-samples/aws-cdk-examples/tree/master/typescript/proxy-apigateway/) | Use ApiGateway to set up a http proxy |
```
Comment (nija-at) [inline README.md:52]: Link is broken.

```suggestion
| [http-proxy-apigateway](https://github.com/aws-samples/aws-cdk-examples/tree/master/typescript/http-proxy-apigateway/) | Use ApiGateway to set up a http proxy |
```

**Comment by nija-at** [comment]
@Mergifyio refresh

**Pull request merged**